openedx/core/djangoapps/programs/utils.py

```diff
@@ -248,6 +249,11 @@ def __init__(self, program_data, user):
         self.course_overview = None
         self.enrollment_start = None
 
+        # Values for programs' price calculation.
+        self.data['avg_price_per_course'] = 0
+        self.data['number_of_courses'] = 0
+        self.data['full_program_price'] = 0
```

Comment:
I think we want to refactor the changes to the `ProgramDataExtender` a bit. Instead of continuing to add to this class, we should create a subclass of `ProgramDataExtender`, call it `ProgramMarketingDataExtender`. This should enable us to override the _extend_course_runs function adding in the logic for collecting the instructors, course price data, and can_enroll value. We should be able to get rid of the `include_instructors` keyword argument in `ProgramDataExtender.extend` if we do this right.

Please take a crack at this approach and ping me if you get stuck.